### PR TITLE
Shanmathi year redux

### PR DESCRIPTION
**Pull request by mathikumaravel**
year get and delete is completed working on post api

### CODE REVIEW

File: src/redux/actions/yearsActions.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import { toast } from "react-toastify";
+import { baseUrl } from "../..";
+import { getAccessToken } from "../../config/getAccessToken";
+import { ActionTypes } from "../constants/action-types";
+
+export const fecthYears = () => {
+    return async function (dispatch, getState) {
+        getAccessToken();
+        const response = await axios.get(`${baseUrl}year`);
+        console.log(response);
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js]: remove console

File: src/redux/actions/yearsActions.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        response.data.data.map((data, index) => {
+            data.index = index + 1;
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js:12]: please check this line

File: src/redux/actions/yearsActions.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+};
+export const addYear = (acdYear) => {
+    return async function (dispatch, getState) {
+        getAccessToken();
+        const response = await axios
+            .post(`${baseUrl}year`, {
+                academic_year: `${acdYear.fromYear}-${acdYear.toYear}`,
+            })
+            .then((response) => {
+                console.log(response.data);
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js]: remove console

File: src/redux/actions/yearsActions.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                        pauseOnHover: true,
+                        draggable: true,
+                        progress: undefined,
+                    });
+                }
+            })
+            .catch((e) => {
+                console.log(e);
+            });
+        console.log(response);
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js]: remove console

File: src/redux/reducers/yearsReducer.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                ...state,
+                products: payload,
+            };
+        default:
+            return state;
+    }
+};
+
+export const selectedYearsReducer = (state = {}, { type, payload }) => {
+    console.log(type);
```
Comment (asainrafik) [inline src/redux/reducers/yearsReducer.js]: remove console

File: src/components/Year/Year.tsx [hunk 6]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
-      history.push(location.pathname);
-    }, 100);
-  }, []);
+    useEffect(() => {
+        setTimeout(() => {
+            console.log(location.pathname);
+            history.push(location.pathname);
+        }, 100);
+    }, []);
+    console.log(year, "122");
```
Comment (asainrafik) [inline src/components/Year/Year.tsx:38]: remove console

File: src/components/Year/Year.tsx [hunk 7]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
+    //                 data.index = index + 1;
+    //             });
+    //             setAllAcademicYear(res.data.data);
+    //             setSpinnerLoad(false);
+    //         })
+    //         .catch((e: any) => {
+    //             console.log(e);
+    //             setSpinnerLoad(false);
+    //         });
+    // };
```
Comment (asainrafik) [inline src/components/Year/Year.tsx]: comment line remove

File: src/components/Year/Year.tsx [hunk 8]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
-          .toString()
-          .toLowerCase()
-          .includes(filter.toString().toLowerCase())
-      );
-    });
+    //Modal Popup
+    const [show, setShow] = useState(false);
+    const handleClose = () => {
+        setShow(false);
+        dispatch(deleteYear(datatoDelete.id, datatoDelete.index));
```
Comment (asainrafik) [inline src/components/Year/Year.tsx:98]: please check the line

File: src/components/Year/Year.tsx [hunk 9]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
+        setFromAcdYear(newDateArr);
+        setAcdYear({
+            fromYear: new Date().getFullYear(),
+            toYear: new Date().getFullYear() + 1,
+        });
+    };
 
-  const deleteAnAcademicYear = (year: any, index: any) => {
-    setSpinnerLoad(true);
+    // const deleteAnAcademicYear = (year: any, index: any) => {
```
Comment (asainrafik) [inline src/components/Year/Year.tsx]: please remove the comment line

File: src/components/Year/Year.tsx [hunk 10]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
-                progress: undefined,
-              });
-            } else {
-              toast.warning("Year Already Added", {
+    const handleSubmit = async (e: any) => {
+        if (acdYear.fromYear && acdYear.fromYear.toString().length > 3) {
+            try {
+                dispatch(addYear());
+                // getAccessToken();
+                // const res: any = await axios
```
Comment (asainrafik) [inline src/components/Year/Year.tsx]: please remove the comment line

File: src/components/Year/Year.tsx [hunk 11]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
+                                            ) : (
+                                                <div>
+                                                    <Container>
+                                                        <Row className="justify-content-md-center">
+                                                            <Col md="6">
+                                                                <Form.Group className="mb-3" controlId="formBasicEmail">
+                                                                    <Form.Label>Academic Year</Form.Label>
+                                                                    <Form.Control
+                                                                        onChange={(e: any) => {
+                                                                            console.log(e.target.value);
```
Comment (asainrafik) [inline src/components/Year/Year.tsx:279]: please remove console

File: src/components/Year/Year.tsx [hunk 12]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
 
-  useEffect(() => {
-    setTimeout(() => {
-      console.log(location.pathname);
-      history.push(location.pathname);
-    }, 100);
-  }, []);
+    useEffect(() => {
+        setTimeout(() => {
+            console.log(location.pathname);
```
Comment (asainrafik) [inline src/components/Year/Year.tsx:34]: remove console

File: src/components/Year/Year.tsx [hunk 13]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
-      newDateArr.push(new Date().getFullYear() + i);
-    }
-    setFromAcdYear(newDateArr);
-    setAcdYear({
-      fromYear: new Date().getFullYear(),
-      toYear: new Date().getFullYear() + 1,
-    });
-  };
+    const callTheYearUpdater = () => {
+        console.log(new Date().getFullYear());
```
Comment (asainrafik) [inline src/components/Year/Year.tsx:125]: remove console

File: src/components/Year/Year.tsx [hunk 14]
```diff
@@ -2,466 +2,438 @@ import { useEffect, useState } from "react";
...
+                //             });
+                //         } else {
+                //             toast.warning("Year Already Added", {
+                //                 position: "top-right",
+                //                 autoClose: 5000,
+                //                 hideProgressBar: false,
+                //                 closeOnClick: true,
+                //                 pauseOnHover: true,
+                //                 draggable: true,
+                //                 progress: undefined,
```
Comment (asainrafik) [inline src/components/Year/Year.tsx]: remove commend line

File: src/redux/actions/yearsActions.js [hunk 15]
```diff
@@ -0,0 +1,100 @@
+import axios from "axios";
+import { toast } from "react-toastify";
+import { baseUrl } from "../..";
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js]: where is coming in baseUrl

**Review by asainrafik — CHANGES_REQUESTED** [review]

File: src/redux/actions/yearsActions.js (hunk 15 above)
Comment (mathikumaravel) [reply src/redux/actions/yearsActions.js]: issue fixed

File: src/components/Year/Year.tsx (hunk 14 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx]: done

File: src/components/Year/Year.tsx (hunk 13 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx:125]: removed

File: src/components/Year/Year.tsx (hunk 12 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx:34]: removed

File: src/components/Year/Year.tsx (hunk 11 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx:279]: removed

File: src/components/Year/Year.tsx (hunk 10 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx]: removed the comment line

File: src/components/Year/Year.tsx (hunk 9 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx]: removed the comment line

File: src/components/Year/Year.tsx (hunk 8 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx:98]: I checked the line here it is used to dispatch the delete action for a specific id and the index of the id.

File: src/components/Year/Year.tsx (hunk 7 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx]: comment lines are removed

File: src/redux/actions/yearsActions.js (hunk 2 above)
Comment (mathikumaravel) [reply src/redux/actions/yearsActions.js:12]: this line is to show the index of data

File: src/redux/actions/yearsActions.js (hunk 3 above)
Comment (mathikumaravel) [reply src/redux/actions/yearsActions.js]: removed

File: src/redux/reducers/yearsReducer.js (hunk 5 above)
Comment (mathikumaravel) [reply src/redux/reducers/yearsReducer.js]: removed the console

File: src/components/Year/Year.tsx (hunk 6 above)
Comment (mathikumaravel) [reply src/components/Year/Year.tsx:38]: removed

File: src/redux/actions/yearsActions.js (hunk 4 above)
Comment (mathikumaravel) [reply src/redux/actions/yearsActions.js]: removed

File: src/redux/actions/yearsActions.js [hunk 16]
```diff
@@ -1,30 +1,29 @@
...
+    try {
+        const res = await axios
             .post(`${baseUrl}year`, {
                 academic_year: `${acdYear.fromYear}-${acdYear.toYear}`,
             })
-            .then((response) => {
-                console.log(response.data);
-                if (response.data.data.insertId) {
+            .then((res) => {
+                console.log(res, "lklk");
```
Comment (asainrafik) [inline src/redux/actions/yearsActions.js]: remove console

**Review by asainrafik — CHANGES_REQUESTED** [review]

File: src/redux/actions/yearsActions.js (hunk 16 above)
Comment (mathikumaravel) [reply src/redux/actions/yearsActions.js]: removed

**Review by asainrafik — APPROVED** [review]

**Review by asainrafik — APPROVED** [review]

**Review by asainrafik — APPROVED** [review]

**Review by asainrafik — APPROVED** [review]

**Review by asainrafik — APPROVED** [review]

**Pull request closed without merging**